diff --git a/components/paint/painter.rs b/components/paint/painter.rs
--- a/components/paint/painter.rs
+++ b/components/paint/painter.rs
@@ -21,9 +21,11 @@ use paint_api::largest_contentful_paint_candidate::LCPCandidate;
 use paint_api::rendering_context::RenderingContext;
 use paint_api::viewport_description::ViewportDescription;
 use paint_api::{
     ImageUpdate, PipelineExitSource, SendableFrameTree, SerializableDisplayListPayload,
-    SerializableImageData, WebRenderExternalImageHandlers, WebRenderImageHandlerType, WebViewTrait,
+    SerializableImageData, WebRenderExternalImageHandlers, WebViewTrait,
 };
+#[cfg(any(feature = "webgl", feature = "webgpu"))]
+use paint_api::WebRenderImageHandlerType;
 use profile_traits::time::{ProfilerCategory, ProfilerChan};
 use profile_traits::time_profile;
 use rustc_hash::{FxHashMap, FxHashSet};
